startup.py

```diff
@@ -0,0 +1,238 @@
+    EXECUTABLE_MATCH_TEMPLATES = {
+        "darwin": [
+            # /Applications/Houdini 15.5.565/Houdini.app
+            "/Applications/Houdini {version}/{product}.app",
+
+            # /Applications/Houdini/Houdini16.0.504.20/Houdini Core 16.0.504.20.app
+            "/Applications/Houdini/Houdini{version}/{product} {version_back}.app",
+        ],
+        "win32": [
+            # C:\Program Files\Side Effects Software\Houdini 15.5.565\bin\houdinifx.exe
+            "C:/Program Files/Side Effects Software/Houdini {version}/bin/{executable}.exe",
+        ],
+        "linux2": [
+            # example path: /opt/hfs14.0.444/bin/houdinifx
+            "/opt/hfs{version}/bin/{executable}",
+        ]
+    }
+
+    @property
+    def minimum_supported_version(self):
+        """The minimum supported Houdini version."""
+        return "12.0"
+
+    def scan_software(self):
+        """
+        Performs a scan for software installations.
+
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+
+        software_versions = []
+
+        for sw_version in self._find_software_versions():
+            if self.is_version_supported(sw_version):
+                self.logger.debug("Accepting %s", sw_version)
+                software_versions.append(sw_version)
+            else:
+                self.logger.debug("Rejecting %s", sw_version)
+                continue
+
+        return software_versions
+
+    def prepare_launch(self, exec_path, args, file_to_open=None):
+        """
+        Prepares the given software for launch
+
+        :param str exec_path: Path to DCC executable to launch
+
+        :param str args: Command line arguments as strings
+
+        :param str file_to_open: (optional) Full path name of a file to open on
+            launch
+
+        :returns: :class:`LaunchInformation` instance
+        """
+
+        tk_houdini_python_path = os.path.join(
+            self.disk_location,
+            "python",
+        )
+
+        sys.path.insert(0, tk_houdini_python_path)
+        from tk_houdini import bootstrap
+
+        # determine all environment variables
+        required_env = bootstrap.compute_environment()
+
+        # Add std context and site info to the env
+        std_env = self.get_standard_plugin_environment()
+        required_env.update(std_env)
+
+        return LaunchInformation(exec_path, args, required_env)
+
+    def _find_software_versions(self):
+        """
+        Scan the filesystem for all houdini executables.
+
+        :return: A list of :class:`SoftwareVersion` objects.
+        """
+
+        self.logger.debug("Scanning for Houdini executables...")
+
+        # use the bundled icon
+        icon_path = os.path.join(
+            self.disk_location,
+            "resources",
+            "houdini_logo.png"
+        )
+        self.logger.debug("Using icon path: %s" % (icon_path,))
+
+        if sys.platform not in self.EXECUTABLE_MATCH_TEMPLATES:
```

Comment:
Changed line `161` to this:

`match_templates = self.EXECUTABLE_MATCH_TEMPLATES.get(sys.platform, [])`